### ISSUE TITLE
Improved option for map rotation

### ISSUE THREAD

**Issue by habitoti** [osmandapp/OsmAnd#14354]
# 🚀 feature request

### Description

Introduce a new option for map rotation

### Describe the solution you'd like

Map rotation is currently configurable for "North up", "Motion based" or "Compass". When going by bike, "Compass" would _wiggle_ the map a lot (for handlebar movement alone), so it's too _sensitive_. Motion based in contrast is perfect and reliable and keeps the map display very steady. However when you come to a stop, it obviously doesn't provide proper orientation any longer. However, on a bike you might have just stopped to checkout where to go next, so having the map being oriented properly would now be beneficial. 

### Describe alternatives you've considered

Introduce a new option "Auto" or "Dynamic" that would switch between " Compass" and "Motion based" depending on speed. Above say "5km/h" it would switch to motion based orientation, below that it would be compass based.

**Comment by tradavyvy** [comment]
Good idea, but 5km/h as a limit would not work while walking. So maybe make it dependent on the current profile or personal preference.

**Comment by habitoti** [comment]
Yes, I was obviously just thinking of cycling. But also, for hiking, "Compass" is probably just fine as option, as you normally only occasionally pick up the phone to check location and track, and then you'll need compass anyhow. But agree, if speed threshold is configurable per profile,this would also be nice.

**Comment by tradavyvy** [comment]
Or maybe something like 'freezing' the orientation to the 'last-significant-direction' when speed is below the treshold, because what i think is most problematic is the random movement of the map because of the fluctuations in gps location when not moving.

**Comment by habitoti** [comment]
I think it actually freezes the last orientation already (maybe because I had stable GPS and not much of "hopping"). 
However what I tend to do when at a some crossing with multiple exits is to slowly turn the handlebar or whole bike to "look" into according directions and align with the map on the display. This obviously just works with the compass then.
Even if threshold would be at 10km/h (to avoid random GPS fluctuation effects at low speeds) it would be great feature for biking, were you usually cruise above that threshold and then can enjoy a steady map display...

**Comment by Jiuvel** [comment]
Hello, yes I think something should be done to solve the problem of the map fluctuations due to noise in the GPS readings.  I always have the map set to "direction of movement" and it works fine while you are moving, the problem occurs when you stop and the map rotates both ways because it is reading small speeds (noise).  I can contribute that I have been using the application for a long time and in previous versions (up to 3.8.4) it was much more stable in that sense, although the problem was not completely eradicated.  A few days ago I opened an issue about it but it was closed.  Greetings.

**Comment by Jiuvel** [comment]
To clarify, I imagine that we are talking about viewing the map without being in navigation mode, because if so, in the navigation options there is the option to select a speed threshold so that the map is oriented in the direction of movement, but Osmand does not it is always used for navigation, so it is important that the orientation of the map in view mode also works correctly.  Greetings.

**Comment by pebogufi** [comment]
Did you have a look on
set Profile, then in Settings
Navigation settings / Map during Navigation / Map orientation threshold
try setting the value to 0

**Comment by habitoti** [comment]
That sounds actually like exactly the feature I was asking for. I was just on 6-day trip with my bike and found that option also when browsing randomly through all the tons of options (I have to say that it's not really easy to find these options at all. Even once I found them, I am almost even more desperate trying to find them again a second time, when I know it's there but I just don't know how I got there in the first place...)
Unfortunately, that option doesn't work, or at least not reliably. The detection of a 0 km/h speed seems to be impossible with GPS on (due to fluctuations) and it almost never switches to compass. If it does (I saw it!), it almost instantly switches back to motion based once I turn the device just a bit. 5km/h sounds like a good option, though, however for some reason it rather stays on compass then for a very long time, even if the speed display already shows a steady 15km/h speed. At some point in time it switches to motion based, only to erratically fall back to compass over and over, though I am steadily cruising at 15-18km/h. I think this feature needs some more tweaking and intelligent heuristics to make any sense at all.

**Comment by sonora** [comment]
Just to be sure: Under the "Configure Profile"'s General settings, you do have "Use Kaiman filter" already set to on, right? Because that takes some noise out of the compass orientation by adding inertia.

**Comment by habitoti** [comment]
Where would I find that filter setting? Tried, but couldn‘t find it (under iOS)

**Comment by sonora** [comment]
I just checked our iOS code, and couldn't really locate it off hand. Checking our documentation here https://osmand.net/docs/user/map/interact-with-map it seems to indicate the feature is so far only available on Android. Chances are that would be another, and smart, way to solve this issue... without an additional option.

**Comment by jojoob** [comment]
> Did you have a look on set Profile, then in Settings Navigation settings / Map during Navigation / Map orientation threshold try setting the value to 0

I desperately searched (because I could swear I already seen it in the past like @habitoti) for this setting but could not find it (latest Android version OsmAnd+ 4.5.8). Was it removed since the map orientation during navigation is now the same as for the profile when not navigating? I'am actually looking for an option to choose a different (default) map orientation during navigation for the same profile. This would be useful for example to have the map North up while browsing the map or reviewing a route for navigation and to have the map orientation automatically switched to movement direction as soon as I start the navigation.

<img src="https://user-images.githubusercontent.com/4180675/257479759-3afe2a04-83b9-475c-ad0f-65fd1d11d39d.jpg" alt="Screenshot_20230801_115032_OsmAnd+" width="350">
